R/lint.R

```diff
@@ -35,8 +35,17 @@ NULL
 #' }
 #'
 #' @export
-lint <- function(filename, linters = NULL, cache = FALSE, ..., parse_settings = TRUE, text = NULL) {
-
+lint <- function(filename, linters = NULL, ..., cache = FALSE, parse_settings = TRUE, text = NULL) {
+  # TODO(next release after 3.0.0): remove this deprecated workaround
+  dots <- list(...)
+  if (length(dots) > 0L && is.logical(dots[[1L]])) {
+    warning(
+      "'cache' is no longer available as a positional argument; please supply 'cache' as a named argument instead. ",
+      "This warning will be upgraded to an error in the next release."
+    )
+    cache <- dots[[1L]]
+    dots <- dots[-1L]
+  }
```

Comment:
It just occurred to me that there might be a way to localize the deprecation code to one spot by using `Recall()`. Not 100% sure though. Either way is fine by me.

Reply:
i'm also not familiar with `Recall()`... better to avoid then :)

IIUC it would still have the issue that we need to strip out `...[[1]]`, so it would look need `do.call(Recall)`, which may not work (I have in mind the "Note" in `?Recall`)

Reply:
Also, is this tested? I only see a test for `lint_dir`.

Reply:
added one